src/index.jsx

```diff
@@ -66,20 +66,23 @@ class IntlCurrencyInput extends Component {
 
   updateValues(event) {
     const [value, maskedValue] = this.calculateValues(event.target.value, this.props.config, this.props.currency)
-
-    this.setState({
-      maskedValue,
-    })
-
-    return [value, maskedValue]
+    if (!this.props.max || value <= this.props.max) {
+      this.setState({
+        maskedValue,
+      })
+
+      return [value, maskedValue]
+    } else {
+      return [this.normalizeValue(this.state.maskedValue)/100, this.state.maskedValue]
```

Comment:
the returned value on the case that the value was over max was also wrong, my mistake. I forgot to divide by 100.
Would you rather use 
`
this.calculateValues(this.maskedValue, this.props.config, this.props.currency)
`
?

Reply:
I believe we should start saving the value state along with maskedValue so we can return the last saved value before the limit constraint was reached, like this: 

```js
updateValues(event) {
    const [value, maskedValue] = this.calculateValues(event.target.value, this.props.config, this.props.currency)
    if (!this.props.max || value <= this.props.max) {
      this.setState({
        value,
        maskedValue,
      })

      return [value, maskedValue]
    } else {
      return [this.state.value, this.state.maskedValue]
    }
  }
```

Does it make sense?

Reply:
@thiagozanetti I'd rather keep a single source of truth on my component's states. Normally I like keeping the value and formatting the string as necessary. In this case I'd keep the string and generate the value when necessary, because of floating point rounding errors.

Reply:
If you do decide to change the way we manage the state, I believe it should be done in a separate PR

Reply:
It seems reasonable to me.